[refs #00209] Heading line heights

penpadstudio opened the pull request:
Updated Headings to correct sizing and added specific line heights. Also added missing H6. Ready to be merged into develop branch on code approval.

Commit:
Line Height and Font size on Headers

Changed font size on H3,4,5 and added line height for all headers

Commit:
Heading Enhancements

Updated Headings to correct sizing and added specific line heights. Also added missing H6.

cehwitham reviewed (commented):
Is  there a need to introduce all the line heights @Pro-Paul ? We didn't have this currently?

penpadstudio commented:
We didn't have the line heights set, they were all generic so they have now been set, I will be doing the same for body copy as they also have not been set according to prior design format.

cehwitham reviewed (changes requested):
Do we not already have a relative method for handling line heights? We shouldn't really be using exact values for line heights as it's not best practice.

https://csswizardry.com/2012/11/code-smells-in-css/#hard-codedabsolute-values

penpadstudio commented:
I might need some help to talk through what needs to be updated and how in the code for this one @cehwitham

cehwitham commented:
Probably one to visit with @csswizardry

csswizardry committed:
Merge branch 'develop' into txt-00209

csswizardry committed:
[refs #00209] Use mixin to apply line-heights

We can save a few lines by using the `font-size` mixin to generate our
`line-height`s for us. Let’s do that!

csswizardry committed:
[refs #00209] Update font-size mixin

I’ve updated the font-size mixin to take explicit pixel values but still
convert them into unitless at compile-time. This means we can
effectively target a line height of, e.g., 28px, but it will still be
transformed into a more friendly unitless value.

cehwitham commented on tools/_tools.mixins.scss:
```diff
@@ -51,13 +51,22 @@
+  // If we haven’t specified a line-height explicitly, let’s automatically
+  // create one that sits on the next closest unit of our baseline grid.
   @if ($line-height == auto) {
     line-height: ceil($font-size / $global-line-height) * ($global-line-height / $font-size);
```
Won't this always = 1? Should one of the variables be a different font size? Global or something?

csswizardry replied:
The `ceil()` rounds up the first equation so it will always be >=1 👍

cehwitham replied:
Ah I'd missed the middle set of brackets.

csswizardry committed:
[refs #00209] Fix regressed SMALL elements

Changes to the font-size mixin regressed a hard-coded line-height set on
SMALLs. Should all be fixed now.

csswizardry committed:
[refs #00209] Fix regressed BLOCKQUOTEs

Changes to the font-size mixin regressed BLOCKQUOTE elements. Now fixed.

csswizardry committed:
[refs #00209] Fix regressed quotes

Again, some quote staying had regressed with the introduction of the new
font-size mixin. Fixed!

csswizardry reviewed (approved):
Worked on this with @Pro-Paul; we’re both happy with it.
